projects/ui/src/components/App/SdkProvider.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { createContext, useMemo } from 'react';
+import { BeanstalkSDK } from '@beanstalk/sdk';
+import { ethers } from 'ethers';
+import { useSigner } from '~/hooks/ledger/useSigner';
+import useChainId from '~/hooks/chain/useChainId';
+import { SupportedChainId, TESTNET_RPC_ADDRESSES } from '~/constants';
+import { BEAN, BEAN_CRV3_LP, CRV3, DAI, ETH, PODS, SPROUTS, UNRIPE_BEAN, UNRIPE_BEAN_CRV3, USDC, USDT, WETH } from '~/constants/tokens';
+
+const useBeanstalkSdkContext = () => {
+  const { data: signer } = useSigner();
+  const chainId = useChainId();
+
+  const sdk = useMemo(() => {
+    let provider: ethers.providers.JsonRpcProvider | undefined;
+    if (chainId === SupportedChainId.MAINNET) {
+      provider = new ethers.providers.AlchemyProvider(
+        chainId,
+        import.meta.env.VITE_ALCHEMY_API_KEY
+      );
+    } else if (TESTNET_RPC_ADDRESSES[chainId]) {
+      provider = new ethers.providers.WebSocketProvider('ws://localhost:8545');
+      // provider = new ethers.providers.JsonRpcProvider(
+      //   TESTNET_RPC_ADDRESSES[chainId]
+      // );
+    } else {
+      provider = undefined;
+    }
+
+    const _provider = new ethers.providers.WebSocketProvider(
+      'ws://localhost:8545'
+    );
+
+    const _sdk = new BeanstalkSDK({
+      signer: signer ?? undefined,
+      provider: _provider,
```

Comment:
FYI here's my implementation https://github.com/BeanstalkFarms/Beanstalk/pull/287/files#diff-a694b8b0c305431e644cd266565c1ac367fde96fd5066815cc39284c2c7d21f1 

Granted, I might be wrong but I believe provider is automatically set to `signer.provider`.